src/hooks/useLogging/useLogging.ts

```diff
@@ -0,0 +1,52 @@
+import { useAuthentication } from '../../services/authentication'
+import { useHostedSession } from '../useHostedSession'
+import { LogSeverity } from './types'
+
+interface UseLoggingHook {
+  log: (message: {}, severity: LogSeverity) => void
+  infoLog: (message: {}) => void
+  warningLog: (message: {}) => void
+  errorLog: (message: {}, error: string | {}) => void
+}
+
+export const useLogging = (): UseLoggingHook => {
+  const authContext = useAuthentication()
+  const { session } = useHostedSession()
+
+  const log = async (
+    params: {},
+    severity: LogSeverity,
+    error?: string | {}
+  ): Promise<void> => {
+    await fetch('/api/log', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        params: { ...params, authContext, session },
+        severity,
+        error,
+      }),
+    })
+  }
+
+  const infoLog = async (params: {}): Promise<void> => {
+    await log(params, 'INFO')
+  }
```

Comment:
Suggestion: add an explicit `event` property (or something to that effect) so that we don't need to rely on the message to compute statistics.